iaf-aspose/src/main/java/nl/nn/adapterframework/extensions/aspose/services/conv/impl/CisConversionServiceImpl.java

```diff
@@ -0,0 +1,199 @@
+/**
+ * 
+ */
+package nl.nn.adapterframework.extensions.aspose.services.conv.impl;
+
+import java.io.BufferedInputStream;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Files;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import org.apache.log4j.Logger;
+import org.apache.tika.mime.MediaType;
+
+import nl.nn.adapterframework.extensions.aspose.ConversionOption;
+import nl.nn.adapterframework.extensions.aspose.services.conv.CisConversionException;
+import nl.nn.adapterframework.extensions.aspose.services.conv.CisConversionResult;
+import nl.nn.adapterframework.extensions.aspose.services.conv.CisConversionService;
+import nl.nn.adapterframework.extensions.aspose.services.conv.impl.convertors.Convertor;
+import nl.nn.adapterframework.extensions.aspose.services.conv.impl.convertors.ConvertorFactory;
+import nl.nn.adapterframework.extensions.aspose.services.util.DateUtil;
+import nl.nn.adapterframework.extensions.aspose.services.util.FileUtil;
+
+/**
+ * @author <a href="mailto:gerard_van_der_hoorn@deltalloyd.nl">Gerard van der
+ *         Hoorn</a> (d937275)
+ *
+ */
+public class CisConversionServiceImpl implements CisConversionService {
+
+	private static final Logger LOGGER = Logger.getLogger(CisConversionServiceImpl.class);
+
+	// location of converted pdf files
+	private String pdfOutputlocation = "";
+
+	private ConvertorFactory convertorFactory;
+
+	private MediaTypeValidator mediaTypeValidator;
+
+	private String fontsDirectory;
+
+	private static AtomicInteger atomicCount = new AtomicInteger(1);
+
+	public CisConversionServiceImpl(String pdfOutputLocation, String fontsDirectory) {
+		this.pdfOutputlocation = pdfOutputLocation;
+		this.setFontsDirectory(fontsDirectory);
+		convertorFactory = new ConvertorFactory(this, pdfOutputlocation);
+		mediaTypeValidator = new MediaTypeValidator(pdfOutputlocation);
+
+	}
+
+	@Override
+	public CisConversionResult convertToPdf(InputStream inputStream, ConversionOption conversionOption) throws IOException {
+		return convertToPdf(inputStream, null, conversionOption);
+	}
+
+	@Override
+	public CisConversionResult convertToPdf(InputStream inputStream, String filename,
+			ConversionOption conversionOption) throws IOException {
+
+		// InputStream should always be available.
+		if (inputStream == null) {
+			throw new IllegalArgumentException("inputStream == null");
+		}
+
+		File tmpFile = null;
+
+		try {
+
+			tmpFile = getUniqueFile();
+			// Create tmp file to prevent processes consuming inputstream
+			Files.copy(inputStream, tmpFile.toPath());
```

Comment:
Is it really necessary to copy the file here? This will be bad for performance.
It should be possible to read a file multiple times concurrently.